backend/apps/ifc_validation/checks/signatures/check_signatures.py

```diff
@@ -0,0 +1,338 @@
+        if ci["content_type"].native == "signed_data":
+            sd = ci["content"]
+            eci = sd["encap_content_info"]
+            data = eci["content"].native or content[: signature.start].encode("ascii")
+
+            for si in sd["signer_infos"]:
+                sid = si["sid"]
+                # match by issuer+serial or SKI
+                if sid.name == "issuer_and_serial_number":
+                    ias = sid.chosen
+                    ias_issuer = set(
+                        f"{v['type'].human_friendly[0]}={v['value'].native}" for vs in ias["issuer"].chosen for v in vs
+                    )
+
+                    if ias_issuer != self.issuer or ias["serial_number"].native != self.serial_number:
+                        continue
+                else:
+                    ski_ext = self.certificate.extensions.get_extension_for_oid(
+                        ExtensionOID.SUBJECT_KEY_IDENTIFIER
+                    ).value.digest
+                    if sid.chosen.native != ski_ext:
+                        continue
+
+                sig_bytes = si["signature"].native
+                hash_algo = si["digest_algorithm"]["algorithm"].native.upper()
+                try:
+                    self.certificate.public_key().verify(
+                        sig_bytes,
+                        data,
+                        padding.PKCS1v15(),
+                        getattr(hashes, hash_algo)(),
+                    )
+                    return True
+                except InvalidSignature:
+                    return False
+
+            return False
+
+    def verify_pkcs1(self, signature, content) -> bool:
+        expected_size = self.certificate.public_key().key_size // 8
+        if len(signature.signature) != expected_size:
+            return False
+            # raise InvalidSignature(
+            #     f"Bad signature length: expected {expected_size} bytes, got {len(signature.signature)}"
+            # )
+        try:
+            self.certificate.public_key().verify(
+                signature.signature,
+                content[: signature.start].encode("ascii"),
+                padding.PKCS1v15(),
+                hashes.SHA256(),
+            )
+            return True
+        except binascii.Error as e:
+            return False
+        except InvalidSignature as e:
+            return False
+
+    def as_dict(self):
+        excluded_fields = ("certificate",)
+
+        def format(k, v):
+            if k in excluded_fields:
+                return None
+            elif isinstance(v, datetime.datetime):
+                return str(v)
+            elif isinstance(v, set):
+                return ",".join(sorted(v))
+            else:
+                return v
+
+        return {
+            k: format(k, getattr(self, k))
+            for k in (f.name for f in fields(self))
+            if format(k, getattr(self, k)) is not None
+        }
+
+
+class ca_bundle:
```

Comment:
CapWords convention.  It would be helpful to be more explicit here.
```suggestion
class CertAuthorityBundle:
```